### DISCUSSION TITLE
idea how to set article layout via api?

### DISCUSSION THREAD

**Discussion by BoxSo** [alexandreelise/j4x-api-examples#4] [Q&A]
followed your code and seems to work nicely. 

I need to set the layout for imported articles too. any idea how to achieve this? Couldn't find any hint at the api documentation though.

**Comment by alexandreelise** [comment]
Hi @BoxSo ,
The Joomla! 4.x Web Services Api mostly mimics and is based internally upon how Joomla! 4.x works. So it mostly should work the same way as a normal Joomla! article work. One hint you might have is to go to your Joomla! 4.x website in the table named ``` content ``` and the column name ``` attribs  ``` . So if you add ``` attribs ``` column in your csv , add ``` attribs ``` key to my latest php example in 
https://github.com/alexandreelise/j4x-api-examples/blob/6d3eaa90e8cffbe153e0e1a218def0d0986a53dd/using-raw-php/smart-add-edit-with-subform-custom-field-support-from-streamed-csv-url.php#L51

It should work. Hopefully it's more clear to you now. One more thing. If you have a question or a suggestion use discussions tab not the issues. Have a great day.
